packages/algob/src/lib/algo-operator.ts

```diff
@@ -151,13 +151,14 @@ export class AlgoOperatorImpl implements AlgoOperator {
 	async waitForConfirmation(
 		txId: string,
 		waitRounds = wtypes.WAIT_ROUNDS
-	): Promise<ConfirmedTxInfo> {
+	): Promise<TxnReceipt> {
 		const pendingInfo = await algosdk.waitForConfirmation(this.algodClient, txId, waitRounds);
 		if (pendingInfo["pool-error"]) {
 			throw new Error(`Transaction Pool Error: ${pendingInfo["pool-error"] as string}`);
 		}
 		if (pendingInfo[confirmedRound] !== null && pendingInfo[confirmedRound] > 0) {
-			return pendingInfo as ConfirmedTxInfo;
+			const data: TxnReceipt = { txID: txId, ...pendingInfo as ConfirmedTxInfo }
```

Comment:
can just return the created object? there is no need to create a variable for that if we just return it

Reply:
yeah we can do that, will update